javagen/src/main/java/com/azure/autorest/util/TemplateUtil.java

```diff
@@ -207,6 +211,21 @@ public static String getTypeReferenceCreation(IType type) {
     public static void writeTypeReferenceStaticVariable(JavaClass classBlock, GenericType type) {
         // see getLongRunningOperationTypeReferenceExpression
 
+        if (!JavaSettings.getInstance().isBranded()) {
+            StringBuilder sb = new StringBuilder();
+            for (IType typeArgument : type.getTypeArguments()) {
+                if (sb.length() > 0) {
+                    sb.append(", ");
+                }
+                sb.append(typeArgument.getClientType().toString()).append(".class");
+            }
+            classBlock.privateStaticFinalVariable(String.format("Type %1$s = new ParameterizedType() {"
+                            + "@Override public Type getRawType() { return " + type.getName() + ".class; }"
+                            + "@Override public Type[] getActualTypeArguments() { return new Type[] { " + sb + " }; }"
+                            + "@Override public Type getOwnerType() { return null; } }",
+                    CodeNamer.getEnumMemberName("TypeReference" + type.toJavaPropertyString())));
+            return;
+        }
         classBlock.privateStaticFinalVariable(String.format("TypeReference<%1$s> %2$s = new TypeReference<%1$s>() {}",
             type, CodeNamer.getEnumMemberName("TypeReference" + type.toJavaPropertyString())));
```

Comment:
Personally prefer a
```
if () {
  return
} else {
  return
}
```
structure.

I assume here for non-branding, we create a `Type` instance inline?

Reply:
For non-branding, we use `ParameterizedType` in `java.lang.reflect` for types with generics. For non-generic types, we can just use the `Model.class` pattern.